packages/web-app-files/src/views/spaces/GenericSpace.vue

```diff
@@ -389,12 +398,42 @@ export default defineComponent({
     ...mapGetters('Files', ['currentFolder', 'totalFilesCount', 'totalFilesSize']),
     ...mapGetters(['user', 'configuration']),
 
+    folderNotFound() {
+      return this.currentFolder === null
+    },
+
     isEmpty() {
       return this.paginatedResources.length < 1
     },
 
-    folderNotFound() {
-      return this.currentFolder === null
+    isRunningOnEos() {
+      return !!this.configuration?.options?.runningOnEos
+    },
+
+    displayFullAppBar() {
+      return !this.displayResourceAsSingleResource
+    },
+
+    displayResourceAsSingleResource() {
+      if (this.paginatedResources.length !== 1) {
+        return false
+      }
+
+      if (
+        this.isRunningOnEos &&
+        (!this.currentFolder.fileId || this.currentFolder.path === this.paginatedResources[0].path)
+      ) {
+        return true
+      }
+      if (
+        isPublicSpaceResource(this.space) &&
+        !this.currentFolder.canUpload({ user: this.user }) &&
```

Comment:
Because (from my understanding) we'd show the table/tiles if the share/link receiver can upload - one of the usability complaints was that users don't understand that they _can't_ upload when seeing a single shared file in the table view. 
It's more of a PO decision, we could also use the `ResourceDetails.vue` and enable the AppBar with create/upload buttons for single files _and_ upload permissions, and then "switch over" to table/tiles once another resource is added. 
Let me know, happy to finalize this either way :)

Reply:
But can you even upload if `paginatedResources[0].isFolder` is false?

Reply:
But, isn't the check of `paginatedResources[0].isFolder` enough? Or do you have any case where you can upload and is not a folder?

Reply:
My understanding was that we only ever show the single file view for files, never for folders. And for a file you will never be able to upload, like @diocas said. So `isPublicSpaceResource(this.space) && !this.paginatedResources[0].isFolder` should be sufficient.